### ISSUE TITLE
vue3-flicking typescript options mismatch

### ISSUE THREAD

**Issue by hiendv** [naver/egjs-flicking#584]
## Description
Hi, the typing seems a little bit off. When trying to run some of your examples with vue3 and typescript setup, it gave me this error
```
Type '{ options: { align: string; circular: boolean; }; }' is not assignable to type 'IntrinsicAttributes & Partial<{}> & Omit<VNodeProps & AllowedComponentProps & ComponentCustomProps, never> & Partial<...> & Omit<...>'.
  Type '{ options: { align: string; circular: boolean; }; }' is missing the following properties from type 'Omit<ExtractProps<FlickingProps> & VNodeProps & AllowedComponentProps & ComponentCustomProps, DefaultKeys<...>>': status, firstPanelSize

130         <Flicking :options="options">
```

```js
const options = { align: 'prev', circular: true }
```

Look like the options definition is incorrect somehow.

## Steps to check or reproduce
```
yarn create vite
# select vue3 typescript
# install @egjs/vue3-flicking
# use the component just like examples
```

**Comment by WoodNeck** [comment]
Hello @hiendv!
Can I get a more detailed example of your environment?
I really wish to fix this but I couldn't get type inference correctly when using Vue SFC in my env(VS Code + Vetur).

Anyway, I think this is because our definition for FlickingProps lacks default properties for `firstPanelSize` and `status` so an error occurred.
That means, I can fix this easily but I would love to ensure it works correctly when using it in SFC.

> FlickingProps.ts
```ts
class FlickingProps {
  viewportTag = prop<string>({ required: false, default: "div" });
  cameraTag = prop<string>({ required: false, default: "div" });
  hideBeforeInit = prop<boolean>({ required: false, default: false });
  firstPanelSize = prop<string>({ required: false });
  options = prop<Partial<FlickingOptions>>({ required: false, default: {} });
  plugins = prop<Plugin[]>({ required: false, default: [] });
  status = prop<Status>({ required: false });
}
```

>FlickingProps.d.ts
```ts
import { FlickingOptions, Plugin, Status } from "@egjs/flicking";
declare class FlickingProps {
    viewportTag: import("vue-class-component").WithDefault<string>;
    cameraTag: import("vue-class-component").WithDefault<string>;
    hideBeforeInit: import("vue-class-component").WithDefault<boolean>;
    firstPanelSize: string;
    options: Partial<FlickingOptions>;
    plugins: import("vue-class-component").WithDefault<Plugin[]>;
    status: Status;
}
export default FlickingProps;
```

**Comment by hiendv** [comment]
Hi, thanks for the instant reply. Please take a look at this reproduction: https://github.com/hiendv/egjs-flicking-584

```
yarn create vite
# vue-ts
yarn add @egjs/vue3-flicking
yarn
yarn build
# or npx vue-tsc --noEmit
```

Node: v14.18.1
OS: Ubuntu 20.04.3 LTS (Focal Fossa)

**Comment by hiendv** [comment]
I also met more problems with typescript typings
```
Type '{ moveType: string; bound: boolean; bounce: number; renderOnlyVisible: boolean; defaultIndex: number; }' is not assignable to type 'Partial<FlickingOptions>'. ​ts:2322
                    Types of property 'moveType' are incompatible.
```

**Comment by WoodNeck** [comment]
@hiendv
Thanks for creating the repo! I could reproduce this issue with the help of that.
I'll fix this issue asap :)

For the `Types of property 'moveType' are incompatible` issue, that happens because Flicking accepts only pre-defined string literals for the option `moveType`.
You can bypass this by ensuring correct typings for the variable `options`

Method 1, I recommend this as it can suggest correct types for the props inside options object.
```ts
import { FlickingOptions } from "@egjs/vue3-flicking";
const options: Partial<FlickingOptions> = { align: 'prev', circular: true, moveType: "freeScroll" };
```

Method 2, use `as const`
```ts
const options = { align: 'prev', circular: true, moveType: "freeScroll" } as const;
```

**Comment by WoodNeck** [comment]
I've just released vue3-flicking `v4.4.1` that fixes this issue, please try that!

**Comment by hiendv** [comment]
Thanks a lot!
